tests/integration/long/test_custom_payload.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2013-2015 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+
+
+try:
+    import unittest2 as unittest
+except ImportError:
+    import unittest
+
+from cassandra.query import ( SimpleStatement, BatchStatement, BatchType)
+from cassandra.cluster import Cluster
+from cassandra.cluster import NoHostAvailable
+
+from tests.integration import use_singledc, PROTOCOL_VERSION, get_cluster, setup_keyspace
+
+def setup_module():
+    """
+    We need some custom setup for this module. All unit tests in this module
+    require protocol >=4. We won't bother going through the setup required unless that is the
+    protocol version we are using.
+    :return:
+    """
+
+    # If we aren't at protocol v 4 or greater don't waste time setting anything up, all tests will be skipped
+    if PROTOCOL_VERSION >= 4:
+        # Don't start the ccm cluster until we get the custom jvm argument specified
+        use_singledc(start=False)
+        ccm_cluster=get_cluster()
+        # if needed stop CCM cluster
+        ccm_cluster.stop()
+        # This will enable the Mirroring query handler which will echo our custom payload k,v pairs back to us
+        jmv_args=[" -Dcassandra.custom_query_handler_class=org.apache.cassandra.cql3.CustomPayloadMirroringQueryHandler"]
+        ccm_cluster.start(wait_for_binary_proto=True, wait_other_notice=True,jvm_args=jmv_args)
+        # wait for nodes to startup
+        setup_keyspace()
+
+
+
+
+
+def tearDownModule():
+    """
+    The rests of the tests don't need our custom payload query handle so stop the cluster so we
+    don't impact other tests
+    :return:
+    """
+
+    ccm_cluster = get_cluster()
+    if ccm_cluster is not None:
+        ccm_cluster.stop()
+
+class CustomPayloadTests(unittest.TestCase):
+
+    def setUp(self):
+        """
+        Test is skipped if run with cql version <4
+
+        """
+        if PROTOCOL_VERSION < 4:
+            raise unittest.SkipTest(
+                "Protocol 4,0+ is required for custom payloads %r"
```

Comment:
Probably should be something like "Native protocol 4.0+ is required for custom payloads, currently using: %r"